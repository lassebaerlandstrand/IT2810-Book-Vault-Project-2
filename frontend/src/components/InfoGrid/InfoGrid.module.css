:where([data-mantine-color-scheme='light']) .paperBackground {
  background-color: var(--mantine-color-gray-0);
}

:where([data-mantine-color-scheme='dark']) .paperBackground {
  background-color: var(--mantine-color-dark-6);
}
