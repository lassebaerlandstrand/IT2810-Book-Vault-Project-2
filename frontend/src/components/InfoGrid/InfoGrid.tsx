import { Grid, Paper, Text } from '@mantine/core';
import { Book } from '@/generated/graphql';
import styles from './InfoGrid.module.css';

type InfoGridProps = {
  book: Book;
};

const InfoGrid = ({ book }: InfoGridProps) => {
  const tableInfos = [
    { header: 'Title', description: book.title },
    { header: 'Author', description: book.authors.join(', ') },
    { header: 'Publisher', description: book.publisher.name },
    { header: 'Genres', description: book.genres.join(', ') },
    { header: 'Rating', description: book.rating },
    { header: 'Pages', description: book.pages },
    { header: 'Format', description: book.bookFormat },
    {
      header: 'Characters',
      description: book.characters || 'No characters registered',
    },
    { header: 'ISBN', description: book.isbn },
    { header: 'Language', description: book.language },
  ];

  if (book.series) {
    tableInfos.push({ header: 'Series', description: book.series });
  }

  return (
    <Paper shadow="sm" radius="md" className={styles.paperBackground} p="xl" my="xl">
      <Grid grow>
        {tableInfos.map((info, index) => (
          <Grid.Col key={index} span={6}>
            <Text fw={700}>{info.header}</Text>
            <Text>{info.description}</Text>
          </Grid.Col>
        ))}
      </Grid>
    </Paper>
  );
};

export default InfoGrid;
