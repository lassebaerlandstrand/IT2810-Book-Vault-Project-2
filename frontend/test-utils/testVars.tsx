import { Book } from '@/generated/graphql';

export const dummyBook: Book = {
  __typename: 'Book',
  authors: [{ __typename: 'Author', name: 'A name' }],
  coverImg: 'https://placehold.co/200x300?text=Cover%20image%20for%20book',
  genres: [{ __typename: 'Genre', name: 'A genre' }],
  id: '1',
  publisher: { __typename: 'Publisher', name: 'A publisher' },
  title: 'A title',
  bookFormat: 'Paper',
  characters: ['A character'],
  description: 'A deep book about deep topics',
  isbn: '999999999',
  language: 'Swedish',
  pages: 1000,
  publishDate: '03/03/03',
  rating: 3.555,
  series: 'Title #1',
  numRatings: 5000,
  ratingsByStars: [1, 5, 10, 7, 1],
};
