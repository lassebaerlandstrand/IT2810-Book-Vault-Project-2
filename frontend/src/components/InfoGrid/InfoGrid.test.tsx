import { render, screen } from '@test-utils';
import { dummyBook } from '../../../test-utils/testVars';
import InfoGrid from './InfoGrid';

const tableInfos = [
  { header: 'Title', description: dummyBook.title },
  { header: 'Author', description: dummyBook.authors.map((author) => author.name).join(', ') },
  { header: 'Publisher', description: dummyBook.publisher.name },
  { header: 'Genres', description: dummyBook.genres.map((genre) => genre.name).join(', ') },
  { header: 'Rating', description: dummyBook.rating.toFixed(1) },
  { header: 'Pages', description: dummyBook.pages },
  { header: 'Format', description: dummyBook.bookFormat },
  {
    header: 'Characters',
    description: dummyBook.characters?.join(', ') ?? 'No characters defined',
  },
  { header: 'ISBN', description: dummyBook.isbn },
  { header: 'Language', description: dummyBook.language },
];

describe('InfoGrid component', () => {
  test('renders the InfoGrid component properly based on input book', () => {
    render(<InfoGrid book={dummyBook} />);

    tableInfos.forEach((info) => {
      expect(screen.getByText(info.header)).toBeInTheDocument();
      expect(screen.getByText(info.description)).toBeInTheDocument();
    });
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<InfoGrid book={dummyBook} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
