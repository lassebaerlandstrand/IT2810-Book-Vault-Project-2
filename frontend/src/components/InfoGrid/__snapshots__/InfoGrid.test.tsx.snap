// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoGrid component > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-cursor-type: pointer;--mantine-font-family: Poppins, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-font-family-headings: Poppins, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-breakpoint-xs: 30em;--mantine-breakpoint-md: 64em;--mantine-breakpoint-xl: 90em;--mantine-breakpoint-xxs: 24em;--mantine-color-dark-blue-0: #155896;--mantine-color-dark-blue-1: #124b81;--mantine-color-dark-blue-2: #0f3f6c;--mantine-color-dark-blue-3: #0c3256;--mantine-color-dark-blue-4: #092641;--mantine-color-dark-blue-5: #06192b;--mantine-color-dark-blue-6: #051321;--mantine-color-dark-blue-7: #030d16;--mantine-color-dark-blue-8: #030a11;--mantine-color-dark-blue-9: #02070b;--mantine-color-beige-0: #fcfaf7;--mantine-color-beige-1: #f9f5ef;--mantine-color-beige-2: #f6f0e7;--mantine-color-beige-3: #f2eade;--mantine-color-beige-4: #eee4d3;--mantine-color-beige-5: #e9ddc8;--mantine-color-beige-6: #e2d1b5;--mantine-color-beige-7: #dac5a1;--mantine-color-beige-8: #d4bd94;--mantine-color-beige-9: #6a5f4a;}:root[data-mantine-color-scheme="dark"]{--mantine-color-dark-blue-text: var(--mantine-color-dark-blue-4);--mantine-color-dark-blue-filled: var(--mantine-color-dark-blue-8);--mantine-color-dark-blue-filled-hover: var(--mantine-color-dark-blue-9);--mantine-color-dark-blue-light: rgba(5, 19, 33, 0.15);--mantine-color-dark-blue-light-hover: rgba(5, 19, 33, 0.2);--mantine-color-dark-blue-light-color: var(--mantine-color-dark-blue-3);--mantine-color-dark-blue-outline: var(--mantine-color-dark-blue-4);--mantine-color-dark-blue-outline-hover: rgba(9, 38, 65, 0.05);--mantine-color-beige-text: var(--mantine-color-beige-4);--mantine-color-beige-filled: var(--mantine-color-beige-8);--mantine-color-beige-filled-hover: var(--mantine-color-beige-9);--mantine-color-beige-light: rgba(226, 209, 181, 0.15);--mantine-color-beige-light-hover: rgba(226, 209, 181, 0.2);--mantine-color-beige-light-color: var(--mantine-color-beige-3);--mantine-color-beige-outline: var(--mantine-color-beige-4);--mantine-color-beige-outline-hover: rgba(238, 228, 211, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-blue-8);--mantine-color-dark-filled: var(--mantine-color-dark-8);--mantine-color-dark-filled-hover: var(--mantine-color-dark-9);--mantine-color-dark-light: rgba(31, 31, 31, 0.1);--mantine-color-dark-light-hover: rgba(31, 31, 31, 0.12);--mantine-color-dark-light-color: var(--mantine-color-dark-8);--mantine-color-dark-outline: var(--mantine-color-dark-8);--mantine-color-dark-outline-hover: rgba(31, 31, 31, 0.05);--mantine-color-gray-filled: var(--mantine-color-gray-8);--mantine-color-gray-filled-hover: var(--mantine-color-gray-9);--mantine-color-gray-light: rgba(52, 58, 64, 0.1);--mantine-color-gray-light-hover: rgba(52, 58, 64, 0.12);--mantine-color-gray-light-color: var(--mantine-color-gray-8);--mantine-color-gray-outline: var(--mantine-color-gray-8);--mantine-color-gray-outline-hover: rgba(52, 58, 64, 0.05);--mantine-color-red-filled: var(--mantine-color-red-8);--mantine-color-red-filled-hover: var(--mantine-color-red-9);--mantine-color-red-light: rgba(224, 49, 49, 0.1);--mantine-color-red-light-hover: rgba(224, 49, 49, 0.12);--mantine-color-red-light-color: var(--mantine-color-red-8);--mantine-color-red-outline: var(--mantine-color-red-8);--mantine-color-red-outline-hover: rgba(224, 49, 49, 0.05);--mantine-color-pink-filled: var(--mantine-color-pink-8);--mantine-color-pink-filled-hover: var(--mantine-color-pink-9);--mantine-color-pink-light: rgba(194, 37, 92, 0.1);--mantine-color-pink-light-hover: rgba(194, 37, 92, 0.12);--mantine-color-pink-light-color: var(--mantine-color-pink-8);--mantine-color-pink-outline: var(--mantine-color-pink-8);--mantine-color-pink-outline-hover: rgba(194, 37, 92, 0.05);--mantine-color-grape-filled: var(--mantine-color-grape-8);--mantine-color-grape-filled-hover: var(--mantine-color-grape-9);--mantine-color-grape-light: rgba(156, 54, 181, 0.1);--mantine-color-grape-light-hover: rgba(156, 54, 181, 0.12);--mantine-color-grape-light-color: var(--mantine-color-grape-8);--mantine-color-grape-outline: var(--mantine-color-grape-8);--mantine-color-grape-outline-hover: rgba(156, 54, 181, 0.05);--mantine-color-violet-filled: var(--mantine-color-violet-8);--mantine-color-violet-filled-hover: var(--mantine-color-violet-9);--mantine-color-violet-light: rgba(103, 65, 217, 0.1);--mantine-color-violet-light-hover: rgba(103, 65, 217, 0.12);--mantine-color-violet-light-color: var(--mantine-color-violet-8);--mantine-color-violet-outline: var(--mantine-color-violet-8);--mantine-color-violet-outline-hover: rgba(103, 65, 217, 0.05);--mantine-color-indigo-filled: var(--mantine-color-indigo-8);--mantine-color-indigo-filled-hover: var(--mantine-color-indigo-9);--mantine-color-indigo-light: rgba(59, 91, 219, 0.1);--mantine-color-indigo-light-hover: rgba(59, 91, 219, 0.12);--mantine-color-indigo-light-color: var(--mantine-color-indigo-8);--mantine-color-indigo-outline: var(--mantine-color-indigo-8);--mantine-color-indigo-outline-hover: rgba(59, 91, 219, 0.05);--mantine-color-blue-filled: var(--mantine-color-blue-8);--mantine-color-blue-filled-hover: var(--mantine-color-blue-9);--mantine-color-blue-light: rgba(25, 113, 194, 0.1);--mantine-color-blue-light-hover: rgba(25, 113, 194, 0.12);--mantine-color-blue-light-color: var(--mantine-color-blue-8);--mantine-color-blue-outline: var(--mantine-color-blue-8);--mantine-color-blue-outline-hover: rgba(25, 113, 194, 0.05);--mantine-color-cyan-filled: var(--mantine-color-cyan-8);--mantine-color-cyan-filled-hover: var(--mantine-color-cyan-9);--mantine-color-cyan-light: rgba(12, 133, 153, 0.1);--mantine-color-cyan-light-hover: rgba(12, 133, 153, 0.12);--mantine-color-cyan-light-color: var(--mantine-color-cyan-8);--mantine-color-cyan-outline: var(--mantine-color-cyan-8);--mantine-color-cyan-outline-hover: rgba(12, 133, 153, 0.05);--mantine-color-teal-filled: var(--mantine-color-teal-8);--mantine-color-teal-filled-hover: var(--mantine-color-teal-9);--mantine-color-teal-light: rgba(9, 146, 104, 0.1);--mantine-color-teal-light-hover: rgba(9, 146, 104, 0.12);--mantine-color-teal-light-color: var(--mantine-color-teal-8);--mantine-color-teal-outline: var(--mantine-color-teal-8);--mantine-color-teal-outline-hover: rgba(9, 146, 104, 0.05);--mantine-color-green-filled: var(--mantine-color-green-8);--mantine-color-green-filled-hover: var(--mantine-color-green-9);--mantine-color-green-light: rgba(47, 158, 68, 0.1);--mantine-color-green-light-hover: rgba(47, 158, 68, 0.12);--mantine-color-green-light-color: var(--mantine-color-green-8);--mantine-color-green-outline: var(--mantine-color-green-8);--mantine-color-green-outline-hover: rgba(47, 158, 68, 0.05);--mantine-color-lime-filled: var(--mantine-color-lime-8);--mantine-color-lime-filled-hover: var(--mantine-color-lime-9);--mantine-color-lime-light: rgba(102, 168, 15, 0.1);--mantine-color-lime-light-hover: rgba(102, 168, 15, 0.12);--mantine-color-lime-light-color: var(--mantine-color-lime-8);--mantine-color-lime-outline: var(--mantine-color-lime-8);--mantine-color-lime-outline-hover: rgba(102, 168, 15, 0.05);--mantine-color-yellow-filled: var(--mantine-color-yellow-8);--mantine-color-yellow-filled-hover: var(--mantine-color-yellow-9);--mantine-color-yellow-light: rgba(240, 140, 0, 0.1);--mantine-color-yellow-light-hover: rgba(240, 140, 0, 0.12);--mantine-color-yellow-light-color: var(--mantine-color-yellow-8);--mantine-color-yellow-outline: var(--mantine-color-yellow-8);--mantine-color-yellow-outline-hover: rgba(240, 140, 0, 0.05);--mantine-color-orange-filled: var(--mantine-color-orange-8);--mantine-color-orange-filled-hover: var(--mantine-color-orange-9);--mantine-color-orange-light: rgba(232, 89, 12, 0.1);--mantine-color-orange-light-hover: rgba(232, 89, 12, 0.12);--mantine-color-orange-light-color: var(--mantine-color-orange-8);--mantine-color-orange-outline: var(--mantine-color-orange-8);--mantine-color-orange-outline-hover: rgba(232, 89, 12, 0.05);--mantine-color-dark-blue-text: var(--mantine-color-dark-blue-filled);--mantine-color-dark-blue-filled: var(--mantine-color-dark-blue-8);--mantine-color-dark-blue-filled-hover: var(--mantine-color-dark-blue-9);--mantine-color-dark-blue-light: rgba(3, 10, 17, 0.1);--mantine-color-dark-blue-light-hover: rgba(3, 10, 17, 0.12);--mantine-color-dark-blue-light-color: var(--mantine-color-dark-blue-8);--mantine-color-dark-blue-outline: var(--mantine-color-dark-blue-8);--mantine-color-dark-blue-outline-hover: rgba(3, 10, 17, 0.05);--mantine-color-beige-text: var(--mantine-color-beige-filled);--mantine-color-beige-filled: var(--mantine-color-beige-8);--mantine-color-beige-filled-hover: var(--mantine-color-beige-9);--mantine-color-beige-light: rgba(212, 189, 148, 0.1);--mantine-color-beige-light-hover: rgba(212, 189, 148, 0.12);--mantine-color-beige-light-color: var(--mantine-color-beige-8);--mantine-color-beige-outline: var(--mantine-color-beige-8);--mantine-color-beige-outline-hover: rgba(212, 189, 148, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 29.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 30em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 63.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 64em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 89.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 90em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 23.99375em) {.mantine-visible-from-xxs {display: none !important;}}@media (min-width: 24em) {.mantine-hidden-from-xxs {display: none !important;}}
  </style>
  <div
    class="_paperBackground_e10025 m_1b7284a3 mantine-Paper-root"
    style="--paper-radius: var(--mantine-radius-md); --paper-shadow: var(--mantine-shadow-sm); margin-block: var(--mantine-spacing-xl);"
  >
    <style
      data-mantine-styles="inline"
    >
      .__m__-r1g{--grid-gutter:var(--mantine-spacing-md);}
    </style>
    <div
      class="m_410352e9 mantine-Grid-root __m__-r1g"
    >
      <div
        class="m_dee7bd2f mantine-Grid-inner"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-r1i{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r1i"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Title
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            A title
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r1m{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r1m"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Author
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            A name
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r1q{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r1q"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Publisher
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            A publisher
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r1u{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r1u"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Publish Date
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            3 March 2003
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r22{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r22"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Genres
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            A genre
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r26{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r26"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Pages
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            1000
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r2a{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r2a"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Format
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            Paper
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r2e{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r2e"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Characters
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            A character
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r2i{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r2i"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            ISBN
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            999999999
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r2m{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r2m"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Language
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            Swedish
          </p>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-r2q{--col-flex-grow:1;--col-flex-basis:50%;--col-max-width:100%;}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-r2q"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="font-weight: 700;"
          >
            Series
          </p>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
          >
            Title #1
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
